Options.py

```diff
@@ -684,6 +684,20 @@ def from_text(cls, text: str) -> Range:
                 return cls.from_any(cls.default)
             else:  # "false"
                 return cls(0)
+        elif (text[0] == "-" and not text[1:].isnumeric()) \
+                or not text.isnumeric():
+            # Handle conditionally acceptable values here rather than in the f-string
+            default = ""
+            truefalse = ""
+            if hasattr(cls, "default"):
+                default = ", default"
+                if cls.range_start == 0 and cls.default != 0:
+                    truefalse = ", \"true\", \"false\""
+            raise Exception(f"Invalid range value {text!r}. Acceptable values are: "
+                            f"<int>{default}, high, low{truefalse}, "
+                            f"random, random-high, random-middle, random-low, "
+                            f"random-range-low-<min>-<max>, random-range-middle-<min>-<max>, "
+                            f"random-range-high-<min>-<max>, or random-range-<min>-<max>")
```

Comment:
```suggestion
            raise Exception(f"Invalid range value {text!r}."
```

Reply:
Being covered in `weighted_range` doesn't really help here unless the user is willing to try again with a wrong value that starts with `random-`. We could move the text to a common place rather than list them twice? But imo if we show the acceptable values in one invalid-value error we should in all invalid-value errors.

Reply:
Thoughts after I changed it?